05.wc/wc.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = define_options
+
+  if stand_alone?
+    files = ARGV
+    file_details = files.to_h { |file| [file, File.read(File.expand_path(file))] }
+    file_details_total = inilialize_file_details_total(options)
+
+    file_details.each do |file_name, file_string|
+      detail_line = create_detail_line(options, file_string)
+      puts "#{output_lines(detail_line)} #{file_name}"
+
+      next if file_details.size == 1
+
+      options.each_key do |option|
+        file_details_total[option] += detail_line[option]
+      end
+    end
+
+    output_total_line(file_details_total) if file_details.size != 1
+  else
+    detail_line = create_detail_line(options, $stdin.to_a.join)
+    puts output_lines(detail_line)
+  end
+end
+
+def create_detail_line(options, file_string)
+  detail_line = {}
+  options.each_key do |option|
+    detail_line[option] = determine_caluculate(option, file_string)
+  end
+  detail_line
+end
```

Comment:
```suggestion
  {
    l: count_line(file_string),
    w: count_word(file_string),
    c: count_bytesize(file_string)
  }.slice(*options.keys)
end
```

ちょっと高等テクニックです（＆若干不要な処理が増えます）が、こんなふうにもできそうです。